### DISCUSSION TITLE
OQRS enchantsments

### DISCUSSION THREAD

**Discussion by R1BLH** [wavelog/wavelog#662] [Ideas]
OQRS now operates in two modes: a combined search for all station locations and a search for a selected station location.

When searching by selected station location, the OQRS prompt page displays the message specified in the settings for that station location. It's OK.

Searching for all station locations displays a message common to all users of a given WaveLog instance's settings, which is not always convenient.

It would be convenient to have a combined search for all station locations for the specified callsign of the wavelog user on that instance (on the OQRS page there is a search for the user's callsign in the table). And to display the message specified by that user in his account settings, not a general message for all users of that wavelog instance.

It would also be nice to allow the visitor to choose the language in which the OQRS page is displayed. And it would be nice to be able to specify not only the theme, but also the language of the OQRS widget in the widget settings.

**Comment by HB9HIL** [comment]
Hello @R1BLH 

for the first things about grouped search and the OQRS message may @AndreasK79 can say something. 

But I can say something about Language: 

If there is no user logged in (which is the case if some other ham is visiting your page) the language is always automatically chosen by the current set browser language. So I assume that your browser language is set to russian. If you visit my public page [https://wavelog.hb9hil.org/visitor/log](https://wavelog.hb9hil.org/visitor/log) you should see this page in russian. If you set your browser language to chinese you will see this page automatically in chinese. If the set browser language is not available in Wavelog the default language is english. This way you can make sure that every ham which is visiting your page see's the text in his native language (if available in Wavelog). The same behaviour occurs for the new OQRS Widget (see [my qrz.com](https://qrz.com/db/hb9hil) page as reference)

**Reply by HB9HIL** [reply]
> [!TIP]
> If you want to test that with your own Wavelog instance to have to be logged **out** and clear cookies. Otherwise you would see the language set in your user settings.

**Reply by R1BLH** [reply]
@HB9HIL it works! Wonderful!
The language question is gone.
